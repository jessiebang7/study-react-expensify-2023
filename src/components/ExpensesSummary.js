import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export class ExpensesSummary extends React.Component {
  render() {
    return (
      <p>Viewing {this.props.expenseCount} expense{this.props.expenseCount>1 &&'s'} totaling {numeral(this.props.expensesTotal/100).format('$0,0.00')} </p>
    )
  }
}

const mapStateToProps = (state) => ( {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
  }
)
export default connect(mapStateToProps)(ExpensesSummary);