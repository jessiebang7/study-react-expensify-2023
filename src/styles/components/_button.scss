.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $m-size;
  text-decoration: none;
  &:hover {
    background-color: darken($blue, 15%);
    color: darken(white, 20%);
  }
}

.button--link {
  background: none;
  &:hover {
    background: none;
  }
}

.button--secondary {
  background: #888888;
}