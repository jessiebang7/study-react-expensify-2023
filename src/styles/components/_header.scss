.header {
  background-color: $blue-dark;
}
.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}
.header__title {
  color: white;
  text-decoration: none;
  text-transform: uppercase;
  h1 {
    margin: 0;
  }
}