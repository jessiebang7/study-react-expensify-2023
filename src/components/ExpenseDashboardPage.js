import React from 'react';
import ExpenseList from './ExpenseList';
import ExpensesSummary from './ExpensesSummary';
import ExpenseFilters from './ExpenseListFilters';

export const ExpenseDashboardPage = () => (
  <div className='ExpenseDashboardPage'>
    <ExpensesSummary />
    <ExpenseFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;