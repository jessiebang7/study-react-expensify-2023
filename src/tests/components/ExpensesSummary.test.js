import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should ExpensesSummary render correctly with 2 expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensestotal={'$94.34'}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should ExpensesSummary render correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensestotal={'$94.34'}/>);
  expect(wrapper).toMatchSnapshot();
});