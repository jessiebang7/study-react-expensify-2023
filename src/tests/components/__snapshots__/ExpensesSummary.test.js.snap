// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should ExpensesSummary render correctly with 1 expense 1`] = `
<p>
  Viewing 
  1
   expense
  <Component />
   totaling 
  $0.00
   
</p>
`;

exports[`should ExpensesSummary render correctly with 2 expenses 1`] = `
<p>
  Viewing 
  2
   expense
  s
   totaling 
  $0.00
   
</p>
`;
